### PR TITLE
Feature/ros node

### PR DESCRIPTION
**Pull request by giorgiocrovetti**

### CODE REVIEW

File: src/ros_utils_py/src/ros_utils_py/ros_node.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    def do_something(self):
+        raise Exception("Not Implemented")
+      
+    def add_publisher(self,topic_name, data_type):
+        self.publishers[topic_name]=rospy.Publisher(topic_name, data_type, queue_size=1)
+    def add_subscriber(self, topic_name, data_type,callback):
+        self.subscribers[topic_name]=rospy.Subscriber(topic_name, data_type, callback)
+    def add_service(self, service_name, Trigger, callback):
+        self.services[service_name]=rospy.Service(service_name, Trigger, callback)
+
```
Comment (mksetaro) [inline src/ros_utils_py/src/ros_utils_py/ros_node.py:26]: missing service server

File: src/ros_utils_py/src/ros_utils_py/ros_node.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            self.main_task()
+            rate.sleep()
+    def do_something(self):
+        raise Exception("Not Implemented")
+      
+    def add_publisher(self,topic_name, data_type):
+        self.publishers[topic_name]=rospy.Publisher(topic_name, data_type, queue_size=1)
+    def add_subscriber(self, topic_name, data_type,callback):
+        self.subscribers[topic_name]=rospy.Subscriber(topic_name, data_type, callback)
+    def add_service(self, service_name, Trigger, callback):
```
Comment (mksetaro) [inline src/ros_utils_py/src/ros_utils_py/ros_node.py:24]: why all services are set making use of **Trigger** type?

File: src/ros_utils_py/test/generic_ros_node_test.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
+import rospy
+import ros_utils_py.ros_node as node
+from std_msgs.msg import String
+
+def main():
+
+    class Talker(node.GenericROSNode):
```
Comment (mksetaro) [inline src/ros_utils_py/test/generic_ros_node_test.py:8]: move declaration outside main

**Review by mksetaro — CHANGES_REQUESTED** [review]
Minor changes

File: src/ros_utils_py/src/ros_utils_py/ros_node.py (hunk 2 above)
Comment (mlia1) [reply src/ros_utils_py/src/ros_utils_py/ros_node.py:24]: changed in data_type

File: src/ros_utils_py/src/ros_utils_py/ros_node.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            rate.sleep()
+    def do_something(self):
+        raise Exception("Not Implemented")
+      
+    def add_publisher(self,topic_name, data_type):
+        self.publishers[topic_name]=rospy.Publisher(topic_name, data_type, queue_size=1)
+    def add_subscriber(self, topic_name, data_type,callback):
+        self.subscribers[topic_name]=rospy.Subscriber(topic_name, data_type, callback)
+    def add_service_client(self, service_name, data_type):
+        self.subscribers[service_name]=rospy.ServiceProxy(service_name, data_type)
```
Comment (mksetaro) [inline src/ros_utils_py/src/ros_utils_py/ros_node.py:25]: fix-> create clients = {}

File: src/ros_utils_py/src/ros_utils_py/ros_node.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    def add_service_server(self, service_name, data_type, callback):
+        self.services[service_name]=rospy.Service(service_name, data_type, callback)
```
Comment (mksetaro) [inline src/ros_utils_py/src/ros_utils_py/ros_node.py:27]: rename in servers

File: src/ros_utils_py/src/ros_utils_py/ros_node.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            rospy.logerr("cannot find publisher for topic: ", topic_name)
+            raise ValueError("publisher does not exist")
+    def get_subscriber(self,topic_name):
+        try:
+            sub=self.subscribers[topic_name]
+            return sub
+        except:
+            rospy.logerr("cannot find subscriber for topic: ", topic_name)
+            raise ValueError("subscriber does not exist")
+    def get_service(self, service_name):
```
Comment (mksetaro) [inline src/ros_utils_py/src/ros_utils_py/ros_node.py:43]: get_service_client & get_server_client

**Review by mksetaro — CHANGES_REQUESTED** [review]

**Review by mksetaro — APPROVED** [review]

**Pull request merged**